### PR TITLE
hotfix 2.1.2

### PR DESCRIPTION
**Pull request by AlexandreJunod**

### CODE REVIEW

**Comment by AlexandreJunod** [comment]
Payment settings work correctly on preprod with : 
- [ ] Multiple forms (no more payment_settings isn't accessible on None)
- [ ] payment_setting enable
- [ ] Tested on forms that didn't worked before fix

Login work correctly with : 
- [ ] Normal user without redirect
- [ ] Normal user with redirect
- [ ] Integrator without redirect
- [ ] Integrator with redirect
- [ ] Pilot without redirect
- [ ] Pilot with redirect

File: geocity/apps/accounts/views.py [hunk 1]
```diff
@@ -225,14 +226,28 @@ def get_success_url(self):
...
-        if "next" in qs_dict:
-            qs_dict.pop("next")
-
-        return (
-            reverse("accounts:profile") + filter_qs
-            if settings.ENABLE_2FA and not self.request.user.totpdevice_set.exists()
-            else url_value
+        is_2fa_disabled = not settings.ENABLE_2FA
+        untrusted_user_without_totpdevice = not (
+            self.request.user.totpdevice_set.exists()
```
Comment (rbovard) [inline geocity/apps/accounts/views.py:231]: You can assign `user_with_totpdevice` before here and use it

**Review by rbovard — CHANGES_REQUESTED** [review]
Have you tried on a non 2FA instance?

I've got this error after login:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.8/dist-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/django/views/generic/base.py", line 103, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/django/utils/decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/django/views/decorators/debug.py", line 92, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/django/utils/decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/django/utils/decorators.py", line 133, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/django/utils/decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/django/views/decorators/cache.py", line 62, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/django/contrib/auth/views.py", line 90, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/django/views/generic/base.py", line 142, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/django/views/generic/edit.py", line 153, in post
    return self.form_valid(form)
  File "/usr/local/lib/python3.8/dist-packages/django/contrib/auth/views.py", line 110, in form_valid
    return HttpResponseRedirect(self.get_success_url())
  File "/code/geocity/apps/accounts/views.py", line 231, in get_success_url
    self.request.user.totpdevice_set.exists()

Exception Type: AttributeError at /account/login/
Exception Value: 'User' object has no attribute 'totpdevice_set'
```

**Review by rbovard — COMMENTED** [review]
Thanks, I've made some successful tests:
- [x] Login on a non 2FA instance
- [x] Submit form with payment
- [x] Submit form without payment

**Pull request closed without merging**